File: Data Structures & Algorithms/sliding-window-maximum/submission-0.java
class Solution {
    public int[] maxSlidingWindow(int[] nums, int k) {
        int[] answer = new int[nums.length-k+1];
        int i = 0;
        int j = k-1;
        while(j<nums.length){
            int max = nums[i];
            for(int l = i+1;l<=j;l++){
                if(nums[l]>max){
                    max = nums[l];
                }
            }
            answer[i] = max;
            i++;
            j++;
        }
        return answer;
    }
}

File: Data Structures & Algorithms/merge-k-sorted-linked-lists/submission-0.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */

class Solution {
public:
    ListNode* mergeKLists(vector<ListNode*>& lists) {
        if(lists.size() == 0){
            return nullptr;
        }
        ListNode* master = lists[0];
        for(int i = 1;i<lists.size();i++){
            ListNode* final = master;
            
            ListNode* answer = new ListNode();
            ListNode* holder = answer;
            ListNode* current = lists[i];
            while(final != nullptr && current != nullptr){
                if(final->val<current->val){
                    answer->next = new ListNode(final->val);
                    final = final->next;
                    answer = answer->next;
                }
                else{
                    answer->next = new ListNode(current->val);
                    current = current->next;
                    answer = answer->next;
                }
            }
            while(final!=nullptr){
                answer->next = new ListNode(final->val);
                final = final->next;
                answer = answer->next;
            }
            while(current!=nullptr){
                answer->next = new ListNode(current->val);
                answer = answer->next;
                current = current->next;
            }
            
            
            master = holder->next;

        }
        return master;
        
    }
};
